tests/src/kakarot/instructions/test_memory_operations.py

```diff
@@ -28,3 +33,73 @@ def test_should_load_a_value_from_memory_with_offset_larger_than_msize(
             cairo_run(
                 "test__exec_mload_should_load_a_value_from_memory_with_offset_larger_than_msize"
             )
+
+    class TestMcopy:
+
+        @given(
+            memory_init_state=lists(
+                integers(min_value=0, max_value=255), min_size=1, max_size=100
+            ),
+            size_mcopy=integers(min_value=1, max_value=100),
+            src_offset_mcopy=integers(min_value=0, max_value=100),
+            dst_offset_mcopy=integers(min_value=0, max_value=100),
+        )
+        def test_should_copy_a_value_from_memory(
+            self,
+            cairo_run,
+            memory_init_state,
+            size_mcopy,
+            src_offset_mcopy,
+            dst_offset_mcopy,
+        ):
+            (evm, memory) = cairo_run(
+                "test__exec_mcopy",
+                memory_init_state=memory_init_state,
+                size_mcopy=int_to_uint256(size_mcopy),
+                src_offset_mcopy=int_to_uint256(src_offset_mcopy),
+                dst_offset_mcopy=int_to_uint256(dst_offset_mcopy),
+            )
+            memory_init_state_expansion = memory_init_state + [0] * (
+                max(src_offset_mcopy, dst_offset_mcopy)
+                + size_mcopy
+                - len(memory_init_state)
+            )
+            segment_to_copy = memory_init_state_expansion[
+                src_offset_mcopy : src_offset_mcopy + size_mcopy
+            ]
+            expected_memory_state = (
+                memory_init_state_expansion[:dst_offset_mcopy]
+                + segment_to_copy
+                + memory_init_state_expansion[dst_offset_mcopy + size_mcopy :]
+            )
+            words_len = (len(expected_memory_state) + 31) // 32
+            expected_memory_state = "".join(
+                [f"{byte:02x}" for byte in expected_memory_state]
+            ) + "00" * (words_len * 32 - len(expected_memory_state))
```

Comment:
```suggestion
            expected_memory_state = memory_init_state + [0] * (
                max(src_offset_mcopy, dst_offset_mcopy)
                + size_mcopy
                - len(memory_init_state)
            )
            expected_memory_state[dst_offset_mcopy: dst_offset_mcopy + size_mcopy] =             memory_expansion[src_offset_mcopy: src_offset_mcopy + size_mcopy]
```

Reply:
Words length calculation needs to be there as serde is serialiazing the memory by words

Reply:
you can incorporate in the previous length calculation

```
(
	                max(src_offset_mcopy, dst_offset_mcopy)
	                + size_mcopy
	                - len(memory_init_state)
	            )
```